### PR TITLE
Add a setting to choose a register allocator algorithm in the new backend

### PR DESCRIPTION
**Pull request by bnjbvr**

### CODE REVIEW

File: cranelift/codegen/meta/src/shared/settings.rs [hunk 1]
```diff
@@ -3,6 +3,35 @@ use crate::cdsl::settings::{SettingGroup, SettingGroupBuilder};
...
+        crate. Please report register allocation bugs to the maintainers of this crate whenever
+        possible.
+
+        Note: this only applies to target that use the MachInst backend. As of 2020-04-17, this
+        means the x86_64 backend doesn't use this yet.
+
+        Possible values:
+
+        - `backtracking` is a greedy, backtracking register allocator as implemented in
+        Spidermonkey's optimizing tier Ionmonkey. It may take more time to allocate registers, but
```
Comment (cfallin) [inline cranelift/codegen/meta/src/shared/settings.rs:20]: s/Ionmonkey/IonMonkey/

File: cranelift/codegen/meta/src/shared/settings.rs [hunk 2]
```diff
@@ -3,6 +3,35 @@ use crate::cdsl::settings::{SettingGroup, SettingGroupBuilder};
...
+        it should generate better code in general, resulting in better throughput of generate code.
```
Comment (cfallin) [inline cranelift/codegen/meta/src/shared/settings.rs:21]: s/generate/generated/

File: cranelift/codegen/meta/src/shared/settings.rs [hunk 3]
```diff
@@ -3,6 +3,35 @@ use crate::cdsl::settings::{SettingGroup, SettingGroupBuilder};
...
+        - `backtracking_checked` is the backtracking allocator with additional self checks that may
+        take some time to run, and thus are disabled by default.
```
Comment (cfallin) [inline cranelift/codegen/meta/src/shared/settings.rs:23]: s/are/is/ (sorry, nitpicky)

Comment (bjorn3) [reply cranelift/codegen/meta/src/shared/settings.rs:23]: "self check**s**" is multiple.

Comment (bnjbvr) [reply cranelift/codegen/meta/src/shared/settings.rs:23]: reformulated to "thus these checks are", to remove all possible doubt

**Comment by bnjbvr** [comment]
CI is failing in ways I don't understand...

**Pull request merged**